src/server/svr_jobfunc.c

```diff
@@ -6148,19 +6251,25 @@ recreate_exec_vnode(job *pjob, char *vnodelist, char *err_msg,
 			NULL,
 			new_exec_host);
 		pjob->ji_modified = 1;
+	} else {
+		log_err(-1, __func__, "new_exec_host is null or empty string");
+		goto recreate_exec_vnode_exit;
 	}
 
-	if (new_exec_host2[0] != '\0') {
+	if (new_exec_host2 && (new_exec_host2[0] != '\0')) {
 
 		(void)job_attr_def[(int)JOB_ATR_exec_host2].at_decode(
 			&pjob->ji_wattr[(int)JOB_ATR_exec_host2],
 			NULL,
 			NULL,
 			new_exec_host2);
 		pjob->ji_modified = 1;
+	} else {
+		log_err(-1, __func__, "new_exec_host2 is null or empty string");
+		goto recreate_exec_vnode_exit;
 	}
 
-	if (new_select[0] != '\0') {
+	if (new_select && (new_select[0] != '\0')) {
```

Comment:
Not wrong, but other places you check (*newselect != '\0')

Reply:
Fixed in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a